### PR TITLE
Rewrite upload from sync to async (Lokalise updated API)

### PR DESCRIPTION
**Pull request by igorDolzh**
Add check to make sure that the job is done for file uploading

They updated `upload` method from sync to async :) It means that during testing I was lucky :) But during final testing I cautch this problem

https://github.com/lokalise/node-lokalise-api#upload-a-file

### CODE REVIEW

**Review by wfortin — APPROVED** [review]

File: lokalise-file-push/lib/main.js [hunk 1]
```diff
@@ -49,28 +49,40 @@ function readLanguageFile(path) {
...
                 });
-                console.log("Uploaded language file: " + filename);
+                let inteval = yield setInterval(() => __awaiter(this, void 0, void 0, function* () {
+                    if (process.status === "finished") {
+                        clearInterval(inteval);
+                        console.log("Uploaded language file: " + filename);
+                        callback();
+                    }
+                    else {
+                        //@ts-ignore
```
Comment (igorDolzh) [inline lokalise-file-push/lib/main.js:74]: They had a bug in TS for this new released feature

File: lokalise-file-push/src/main.ts [hunk 2]
```diff
@@ -37,34 +37,49 @@ async function uploadFiles({
...
+          clearInterval(inteval);
+          console.log("Uploaded language file: " + filename);
+          callback();
+        } else {
+          //@ts-ignore
+          process = await lokalise.queuedProcesses.get(process.process_id, {
+            project_id: projectId,
+          });
+        }
+      }, 1000);
```
Comment (igorDolzh) [inline lokalise-file-push/src/main.ts:75]: This is the main change

**Comment by igorDolzh** [comment]
This PR is huge because I upgraded `@lokalise/node-api` to the latest version and we keep `node_modules`

File: lokalise-file-push/lib/main.js (hunk 1 above)
Comment (wfortin) [reply lokalise-file-push/lib/main.js:74]: Add it as a comment for the ts ignore

File: lokalise-file-push/lib/main.js [hunk 3]
```diff
@@ -79,9 +91,7 @@ uploadFiles({
...
     tags: JSON.parse(tags),
     locales: JSON.parse(locales),
-})
-    .then(() => {
-    ghCore.setOutput("uploaded", "true");
-    console.log("Finished");
-})
-    .catch((error) => ghCore.setFailed(error ? error.message : "Unknown error"));
+    callback: () => {
+        console.log("Finished");
```
Comment (wfortin) [inline lokalise-file-push/lib/main.js:95]: no success state here?

**Review by pekala — APPROVED** [review]
I went through the 2 million LOC and it's legit! Good job

File: lokalise-file-push/lib/main.js (hunk 3 above)
Comment (igorDolzh) [reply lokalise-file-push/lib/main.js:95]: it is a successful callback and inside there is a catch for errors

**Pull request merged**